Source/Filter/BilinearFilter.cpp

```diff
@@ -63,15 +63,15 @@ void BilinearFilter::ComputeStencils(){
     stencil_x.resize( 1 + npass_each_dir[0] );
     stencil_y.resize( 1 + npass_each_dir[1] );
     stencil_z.resize( 1 + npass_each_dir[2] );
-    compute_stencil(stencil_x, npass_each_dir[0]);
-    compute_stencil(stencil_y, npass_each_dir[1]);
-    compute_stencil(stencil_z, npass_each_dir[2]);
+    compute_stencil(stencil_x, static_cast<unsigned int>(npass_each_dir[0]));
```

Comment:
looking at this and also the `resize()` above, I think that even `npass_each_dir` should be unsigned :D

Reply:
I agree (all those `static_cast` are ugly). `npass_each_dir` is an `amrex::IntVec` . I am not totally sure about how to replace it with an unsigned version, but maybe it can just be an `amrex::Array<unsigned int, AMREX_SPACEDIM>`